pandora-server-directory/src/account/accountRelationship.ts

```diff
@@ -99,6 +99,48 @@ export class AccountRelationship {
 		return false;
 	}
 
+	/**
+	 * Checks if this account's profile is visible to the passed account
+	 * @param queryingAccount - The account attempting to access this account's profile
+	 * @returns - If the access should be allowed
+	 */
+	public async profileVisibleTo(queryingAccount: Account): Promise<boolean> {
+		// Player can always see their own profile
+		if (this.account.id === queryingAccount.id)
+			return true;
+
+		// Moderators can see anyone's profile
+		if (queryingAccount.roles.isAuthorized('moderator'))
```

Comment:
unrelated to this PR, just an idea

maybe we could add a state flag to the account that toggles moderator functionality \
so if you just play normally, you would still be affected by blocks and such, but \
if you have a moderator role, you can enable settings that let you bypass such limitations

Reply:
I'm thinking about simply moving the moderator bits into the existing `Management` menu and putting any moderation tools there. I don't think it is a good idea to allow moderators to have any extra "IC" powers at all - only account-based ones.
If people get stuck we don't want to help them anyway - what would then be point of Pandora? And other cases where moderator would interfere with existing tools I think are all about (at least partially) banning someone.

Reply:
What I meant is to only allow moderators to see user profiles to which they have no access otherwise if they truly wish to do this as a moderator, so a limitation to this exact functionality, not any "IC" powers

(feel free to close this conversation)